PowerEditor/src/Notepad_plus.h

```diff
@@ -636,4 +636,47 @@ friend class FileManager;
 
 	void monitoringStartOrStopAndUpdateUI(Buffer* pBuf, bool isStarting);
 	void updateCommandShortcuts();
+	
+	void handleEnvironmentSettingChange(LPARAM lParam);
+	bool _isGlobalEnvChanged = false;
+
+	typedef struct st_MenuCommandMark {
+		st_MenuCommandMark(UINT uMenuCommandId, const TCHAR* szMarkStr, const char* szTooltipStrId, const TCHAR* szTooltipDefaultStr)
+		{
+			::ZeroMemory(this, sizeof(st_MenuCommandMark));
+			_uMenuCommandId = uMenuCommandId;
+			_szMarkStr = szMarkStr;
+			_clrfMark = CLR_INVALID;
+			_alphaOpacity = 255;
+			_szTooltipStrId = szTooltipStrId;
+			_szTooltipDefaultStr = szTooltipDefaultStr;
+			_uMaxTooltipWidth = (UINT)300;
+			_tiTooltip.cbSize = sizeof(TOOLINFO);
+			_tiTooltip.uFlags = TTF_IDISHWND | TTF_ABSOLUTE | TTF_TRACK | TTF_CENTERTIP;
+		}
+		UINT _uMenuCommandId;				// definitions are in the menuCmdID.h
+		const TCHAR* _szMarkStr;			// single char like "!" or a very short string (the menuitem check-area is limited...)
+		COLORREF _clrfMark;					// foreground color of the above mark, use the CLR_INVALID here to use the current foreground menucolor
+		BYTE _alphaOpacity;					// 0 ... transparent, 255 ... opaque
+		SIZE _sizeMarkArea;					// used at runtime
+		const char* _szTooltipStrId;		// optional, it can be NULL (this is string ID for the NativeLangSpeaker)
+		const TCHAR* _szTooltipDefaultStr;	// optional, it can be NULL
+		HWND _hTooltip;						// used at runtime
+		TOOLINFO _tiTooltip;				// used at runtime
+		UINT _uMaxTooltipWidth;				// multiline wrapping point for longer tooltip strings
+	} MENUCOMMANDMARK, *LPMENUCOMMANDMARK;

```

Comment:
About the structure `st_MenuCommandMark`:
1. Coding standard of Notepad++ is not respected: the "type prefix" of variable naming is not used in Notepad++ code, whereas it's everywhere in this PR.
2. In modern C++ structure (no `typedef`), all the member variables can be initialized directly in the structure - it will improve the code readability.